### ISSUE TITLE
Exception not thrown when user cancel the auth flow in android

### ISSUE THREAD

**Issue by hackrx** [aws-amplify/amplify-flutter#819]
**Describe the bug**
When a user tries loging with `signinWithWebUI` and then press the cross button in the left right corner, then there is no exception is thrown on flutter side in android. In IOS it is working fine.

**To Reproduce**
1. Try signInWithWebUI and google as identity provider.
2. Click on cross button 
3. It will log `D/AuthClient(16746): Authorization flow canceled by user`
4. But we can't catch it on the flutter side.
5. This is causing the loading overlay to continue showing and it is an annoying behaviour.

**Expected behavior**
An exception should be thrown for android case so that we can use analytics and further operation after auth cancel.

**Screenshots**
![image](https://user-images.githubusercontent.com/54453778/129984025-2a2ca788-f247-4f98-872f-9483532a9a63.png)

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[x] Android
[] iOS

 
**Additional context**
```
  amplify_api: ^0.2.0
  amplify_flutter: ^0.2.1
  amplify_auth_cognito: ^0.2.0
```

**Comment by dnys1** [comment]
This should be resolved in version 0.2.8 - @hackrx can you confirm?

**Comment by dnys1** [comment]
Closing for now, since this should be resolved. Please let us know if you continue to experience this issue.
